app/src/main/java/org/thoughtcrime/securesms/onboarding/messagenotifications/MessageNotificationsViewModel.kt

```diff
@@ -78,7 +83,8 @@ internal class MessageNotificationsViewModel(
 
     data class UiState(
         val pushEnabled: Boolean = true,
-        val showDialog: Boolean = false,
+        val showBackWarningDialogForNewAccount: Boolean = false,
+        val showBackWarningDialogForLoad: Boolean = false,
```

Comment:
Why not integrate it in the `showDialog`  instead? If we ever get more ways to show the message, this will become unruly.
`showDialog` could be a nullable string, where the dialog is hidden when the string is null and otherwise displays whatever the string is set to.
Also it seems that with this new change `showBackWarningDialogForNewAccount` is never set to true?